src/Service/VendorService/OverDrive/OverDriveMagazinesVendorService.php

```diff
@@ -0,0 +1,221 @@
+        try {
+            do {
+                $this->progressMessage('Search data well for: "'.self::VENDOR_SEARCH_TERM.'" (Offset: '.$offset.')');
+
+                // Search the data well for material with acSource set to "ereolen magazines".
+                [$pidArray, $more, $offset] = $this->searchService->search(self::VENDOR_SEARCH_TERM, $offset);
+
+                // Get the OverDrive APIs title urls from the results
+                $pidArray = $this->getTitleUrls($pidArray);
+
+                // Get the OverDrive APIs crossRefIds from the results
+                $pidArray = $this->getTitleIds($pidArray);
+
+                // Get the OverDrive cover urls
+                $pidArray = $this->getCoverUrls($pidArray);
+
+                $batchSize = \count($pidArray);
+                $this->updateOrInsertMaterials($pidArray, IdentifierType::PID, $batchSize);
+
+                $this->progressMessageFormatted($this->totalUpdated, $this->totalInserted, $this->totalIsIdentifiers);
+                $this->progressAdvance();
+
+                if ($this->limit && $this->totalIsIdentifiers >= $this->limit) {
+                    $more = false;
+                }
+            } while ($more);
+
+            return VendorImportResultMessage::success($this->totalIsIdentifiers, $this->totalUpdated, $this->totalInserted, $this->totalDeleted);
+        } catch (\Exception $exception) {
+            return VendorImportResultMessage::error($exception->getMessage());
+        }
+    }
+
+    /**
+     * Set config from service from DB vendor object.
+     *
+     * @throws UnknownVendorServiceException
+     * @throws IllegalVendorServiceException
+     */
+    private function loadConfig(): void
+    {
+        $libraryAccountEndpoint = $this->getVendor()->getDataServerURI();
+        $clientId = $this->getVendor()->getDataServerUser();
+        $clientSecret = $this->getVendor()->getDataServerPassword();
+
+        $this->apiClient->setCredentials($libraryAccountEndpoint, $clientId, $clientSecret);
+    }
+
+    /**
+     * Get the title urls for all results in array.
+     *
+     * @param array $pidArray
+     *   An array of pid => result to be converted
+     *
+     * @return array
+     *   Array of pid => title url
+     */
+    private function getTitleUrls(array $pidArray): array
+    {
+        $result = [];
+        foreach ($pidArray as $pid => $item) {
+            $result[$pid] = $this->getTitleUrlFromResult($item['record']['identifier']);
+        }
+
+        return $result;
+    }
+
+    /**
+     * Get the OverDrive title id for all results in array.
+     *
+     * @param array $pidArray
+     *   An array of pid => title url to be converted
+     *
+     * @return array
+     *   Array of pid => title id
+     */
+    private function getTitleIds(array $pidArray): array
+    {
+        $result = [];
+        foreach ($pidArray as $pid => $titleUrl) {
+            $result[$pid] = $this->getTitleIdFromUrl($titleUrl);
```

Comment:
The `getTitleIdFromUrl` might return `null` what happens then?

Reply:
Added null filtering